cobc/typeck.c

```diff
@@ -6269,7 +6284,7 @@ decimal_compute (const int op, cb_tree x, cb_tree y)
 /**
  * expand tree x to the previously allocated decimal tree d
  */
-static void
+static cb_tree
```

Comment:
Actually, `decimal_expand` returns an error node if an error occurs, so it does not always return d/t. I think it is better to keep this approach than returning an integer that has to be tested on every call site to finally replace d/t by an error node at the caller.

Reply:
I see, there is now a single instance of this. If it is needed, then keep this (just mention the "returns x or cb_error_node" part to the function doc comment); I _think_ with this implementation we could get into the error case first, then get in to the CONST case then ABORT... if this is the case we'd need to add the check for d != error_node before the zero check.

I haven't checked - if we get away with just leaving d unchanged then this would be the easier solution and the lesser diff.

Reply:
I updated the comment. Note that the modification I made in this function only applies in a case that had a completely undefined behavior, i.e. the code believed the expression was numeric when it was actually boolean. So, it is not clear what would happen by keeping d instead of the error node. To make the code a bit safer, I added a `error_statement = current_statement;` as it looks like some of the functions test these variables to avoid doing anything with the expressions.